central/resourcecollection/service/service_impl.go

```diff
@@ -164,6 +159,42 @@ func (s *serviceImpl) DeleteCollection(ctx context.Context, request *v1.Resource
 	return &v1.Empty{}, nil
 }
 
+func (s *serviceImpl) checkCollectionReferences(ctx context.Context, request *v1.ResourceByID) error {
+	parentCollectionsQuery := search.NewQueryBuilder().AddExactMatches(search.EmbeddedCollectionID, request.GetId()).ProtoQuery()
+	referencedCollections, err := s.datastore.SearchCollections(ctx, parentCollectionsQuery)
+	if err != nil {
+		return errors.Wrap(err, "Failed to check for collection references")
+	}
+
+	if len(referencedCollections) != 0 {
+		var names []string
+		for _, referencedCollection := range referencedCollections {
+			names = append(names, referencedCollection.GetName())
+		}
+		return errors.Wrapf(errox.ReferencedByAnotherObject, "Collection is in use by one or more other collections by following collections: %q", strings.Join(names, ", "))
```

Comment:
@md2119 Changed the error message. Yeah it can be cumbersome, eventually the error message should link to an Collection consumers, for now this is the easiest to improve the current UX.